src/integration/cluster_integration_tests.md

```diff
@@ -0,0 +1,210 @@
+* Local development
+
+### When Not to Use
+There are some occasions where cluster integration tests may not be the best tool. Consider other options if:
+
+* Test requires you to manipulate the clock.
+   * This is currently unsupported. Component-specific integration tests that are configured programmatically may be a better option. Here are some examples for [dbnodes](https://github.com/m3db/m3/tree/master/src/dbnode/integration)
+* Test requires changing options not exposed by config.
+    * Since cluster integration tests start M3 components via the same entry point as the actual binary, all configuration has to be done via the component configuration file or public APIs. If you need to manipulate some options that are not exposed this way, then these tests may not be a good fit.
+
+## Example Usage
+
+Let's walk through a few examples that outline the most common use cases.
+
+### Denoting test as a cluster integration test
+Any test meant to be considered as a cluster integration test should be tagged with the correct build tag:
+
+```golang
+// +build cluster_integration
+```
+
+### Spinning up a cluster
+
+
+Below is an example that will spin up a cluster with a coordinator, DB node, and aggregator that can interact with each other. It also creates you an unaggregated and an aggregated namespace that you can read and write to.
+
+```golang
+import (
+	"github.com/m3db/m3/src/integration/resources"
+	"github.com/m3db/m3/src/integration/resources/inprocess"
+)
+
+// The {} represent empty configs which will start each component
+// with the default configuration
+cfgs, _ := inprocess.NewClusterConfigsFromYAML(
+	`db: {}`, `{}`, `{}`,
+)
+
+m3, _ = inprocess.NewCluster(cfgs,
+	resources.ClusterOptions{
+		DBNode: resources.NewDBNodeClusterOptions(),
+		Aggregator: resources.NewAggregatorClusterOptions()
+	},
+)
+```
+
+Naturally, you can spin the cluster up with whatever configuration you like. Also, you can use `DBNodeClusterOptions` and `AggregatorClusterOptions` to spin up more interesting clusters. For example:
+
+```golang
+m3, _ := inprocess.NewCluster(configs, resources.ClusterOptions{
+	DBNode: &resources.DBNodeClusterOptions{
+		RF:                 3,
+		NumInstances:       1,
+		NumShards:          4,
+		NumIsolationGroups: 3,
+	},
+	Aggregator: &resources.AggregatorClusterOptions{
+		RF:                 2,
+		NumShards:          4,
+		NumInstances:       2,
+		NumIsolationGroups: 2,
+	}
+})
+```
+This configuration will spin up the following:
+
+* DB nodes with an RF of 3 and 1 instance for each RF (i.e. 3 separate invocations of M3DB in a single process)
+* 1 coordinator
+* Aggregators with an RF of 2 and 2 instances for each RF (i.e. 4 separate aggregator invocations in a single process)
+
+It's worth pointing out that you aren't required to spin up a full cluster each time. `inprocess.NewCluster` also allows you to just spin up a dbnode and coordinator, and later we demonstrate how to spin up just a single component.
+
```

Comment:
do ou want to mention ClusterSpecification in this section?

Reply:
I was going to leave that out for now as I think the use case is a bit niche (e.g. multi-cluster setups). If we get more questions about it we can definitely add.